io.openems.backend.alerting/src/io/openems/backend/alerting/handler/SumStateHandler.java

```diff
@@ -0,0 +1,200 @@
+			MinuteTimer.getInstance().unsubscribe(this.initMetadata);
+		}
+		this.initMetadata = null;
+		this.mss.unregister(this);
+		this.msgScheduler = null;
+		this.mss = null;
+	}
+
+	@Override
+	public void send(ZonedDateTime sentAt, List<SumStateMessage> pack) {
+		// Ensure Edge is still in error state before sending mail.
+		pack.removeIf((msg) -> !this.isEdgeError(msg.getEdgeId()));
+
+		var params = JsonUtils
+				.generateJsonArray(pack.stream().filter(m -> m.shouldSend()).map(SumStateMessage::getParams).toList());
+		if (!params.isEmpty()) {
+			this.mailer.sendMail(sentAt, SumStateMessage.TEMPLATE, params);
+		}
+
+		var logStrBuilder = new StringBuilder(pack.size() * 64);
+		pack.forEach(msg -> {
+			if (msg.shouldSend()) {
+				logStrBuilder.append(msg).append(", ");
+			}
+			this.tryReschedule(msg);
+		});
+		var logStr = logStrBuilder.toString();
+		if (!logStr.isBlank()) {
+			this.log.info("Sent ErrorEdgeMsg: " + logStr);
+		}
+	}
+
+	private void tryReschedule(SumStateMessage msg) {
+		if (msg.update()) {
+			this.msgScheduler.schedule(msg);
+		}
+	}
+
+	private boolean isEdgeError(String edgeId) {
+		var sumState = this.metadata.getSumState(edgeId);
+		if (sumState.isPresent()) {
+			return sumState.get().isAtLeast(Level.FAULT);
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Add Edge to list, with calculated TimeStamp (at which to notify).
+	 *
+	 * @param edge     to add
+	 * @param sumState of edge
+	 * @return {@link OfflineEdgeMessage} generated from edge
+	 */
+	protected SumStateMessage getEdgeMessage(Edge edge, Level sumState) {
+		if (edge == null || edge.getId() == null) {
+			this.log.warn("Called method getEdgeMessage with " //
+					+ edge == null ? "Edge{null}" : "Edge{id=null}");
+		}
+		var message = new SumStateMessage(edge.getId(), sumState, ZonedDateTime.now());
+		return message;
+	}
+
+	private boolean containsEdge(String edgeId) {
+		return this.msgScheduler.isScheduled(msg -> {
+			return Objects.equals(msg.getEdgeId(), edgeId);
+		});
+	}
+
+	protected void tryRemoveEdge(Edge edge) {
+		this.msgScheduler.remove(edge.getId());
+	}
+
+	protected void tryAddEdge(Edge edge, Level sumState) {
+		var msg = this.getEdgeMessage(edge, sumState);
+		if (msg != null) {
+			this.log.debug("Schedule : " + msg);
+			this.msgScheduler.schedule(msg);
+		}
+	}
+
+	@Override
+	public Consumer<EventReader> getEventHandler(String eventTopic) {
+		switch (eventTopic) {
+		case Edge.Events.ON_SET_SUM_STATE: {
+			return (event) -> {
+				final var edgeId = (String) event.getProperty(Edge.Events.OnSetSumState.EDGE_ID);
+				final var level = (Level) event.getProperty(Edge.Events.OnSetSumState.SUM_STATE);
+
+				if (this.testFilter(edgeId, level)) {
+					return;
+				}
+
+				final var edgeOpt = this.metadata.getEdge(edgeId);
+				edgeOpt.ifPresent(edge -> {
+					if (level.isAtLeast(Level.FAULT)) {
+						this.tryAddEdge(edge, level);
+					} else {
+						this.tryRemoveEdge(edge);
+					}
```

Comment:
Hey @da-Kai 

Thank you for providing your wip. 
I was able to compare your implementation to mine and I got a good idea of your intention with your implementation.

I know that you stated in your general code you will "Improved data structure for growing alerting requirements and settings"
I just wanted to ask if the line 140
```java
level.isAtLeast(Level.FAULT) 
```
will be configurable in future iterations (so that different users can be notified on different sumStates)?

Thank you in advance.

Reply:
Hey @DerStoecki 

I have inserted this filter for test purposes only. Later this will be configurable.